### PR TITLE
feat: initialized blocklist and base state

### PR DESCRIPTION
**Pull request by Nilay27**
* initialized Blocklist with nominee set in the constructor.
* Also added `_blockedAccountsIndex`, which will be used when resetting the blocklist by incrementing the index.
* Following nested mapping is used to keep track of the blocklist:
 `mapping(uint256 => mapping(address => bool)) private _blockedAccounts;`
* Tests for covering `isAccountBlocked` will be added in the subsequent PR when inside the tests for setting up the blocklist.

### CODE REVIEW

**Review by ramenforbreakfast — APPROVED** [review]
LGTM

File: apps/smart-contracts/token/contracts/Blocklist.sol [hunk 1]
```diff
@@ -5,9 +5,26 @@ import "./interfaces/IBlocklist.sol";
...
+  }
 
-  function reset(address[] calldata _blockedAccounts) external override onlyOwner {}
+  function set(address[] calldata _accounts, bool[] calldata _blocked)
+    external
+    override
+    onlyOwner
+  {}
+
+  function reset(address[] calldata _newBlockedAccounts) external override onlyOwner {}
```
Comment (ramenforbreakfast) [inline apps/smart-contracts/token/contracts/Blocklist.sol:21]:
```suggestion
  function reset(address[] calldata _accountsToBlock) external override onlyOwner {}
```
Also I was reviewing PR #220 and notice that the parameter naming for the reset() functions is not consistent. I will recommend revised names for them in that PR.

File: apps/smart-contracts/token/test/Blocklist.test.ts [hunk 2]
```diff
@@ -1,5 +1,41 @@
...
+      expect(await blocklist.getNominee()).to.eq(owner.address)
+    })
+
+    it('sets owner to deployer', async () => {
+      expect(await blocklist.owner()).to.eq(deployer.address)
+    })
+
+    it('sets blocked accounts index to zero', async () => {
+      expect(await blocklist.getBlockedAllountsIndex()).to.eq(0)
+    })
```
Comment (XavierEkkel) [inline apps/smart-contracts/token/test/Blocklist.test.ts:39]:
```suggestion
```

File: apps/smart-contracts/token/contracts/Blocklist.sol [hunk 3]
```diff
@@ -5,9 +5,26 @@ import "./interfaces/IBlocklist.sol";
...
+
+  function isAccountBlocked(address _account) external view override returns (bool) {
+    return _blockedAccounts[_blockedAccountsIndex][_account];
+  }
+
+  function getBlockedAllountsIndex() external view override returns (uint256) {
+    return _blockedAccountsIndex;
+  }
```
Comment (XavierEkkel) [inline apps/smart-contracts/token/contracts/Blocklist.sol:29]:
```suggestion
```

File: apps/smart-contracts/token/contracts/Blocklist.sol [hunk 4]
```diff
@@ -5,9 +5,26 @@ import "./interfaces/IBlocklist.sol";
 import "prepo-shared-contracts/contracts/SafeOwnable.sol";
 
 contract Blocklist is IBlocklist, SafeOwnable {
-  constructor(address _nominatedOwner) {}
+  uint256 private _blockedAccountsIndex;
+  mapping(uint256 => mapping(address => bool)) private _blockedAccounts;
```
Comment (XavierEkkel) [inline apps/smart-contracts/token/contracts/Blocklist.sol:9]:
```suggestion
  mapping(uint256 => mapping(address => bool)) private _resetIndexToAccountToBlocked;
```

File: apps/smart-contracts/token/contracts/interfaces/IBlocklist.sol [hunk 5]
```diff
@@ -6,4 +6,8 @@ interface IBlocklist {
   function set(address[] calldata accounts, bool[] calldata blocked) external;
 
   function reset(address[] calldata blockedAccounts) external;
+
+  function isAccountBlocked(address account) external view returns (bool);
```
Comment (XavierEkkel) [inline apps/smart-contracts/token/contracts/interfaces/IBlocklist.sol:10]:
```suggestion
  function isBlocked(address account) external view returns (bool);
```

File: apps/smart-contracts/token/contracts/Blocklist.sol [hunk 6]
```diff
@@ -5,9 +5,26 @@ import "./interfaces/IBlocklist.sol";
 import "prepo-shared-contracts/contracts/SafeOwnable.sol";
 
 contract Blocklist is IBlocklist, SafeOwnable {
-  constructor(address _nominatedOwner) {}
+  uint256 private _blockedAccountsIndex;
```
Comment (XavierEkkel) [inline apps/smart-contracts/token/contracts/Blocklist.sol:8]:
```suggestion
  uint256 private _resetIndex;
```

**Review by XavierEkkel — APPROVED** [review]
LGTM after comments!

**Pull request merged**